2016-10-30  Xuemin He  <xuemin@lnxsrv05.seas.ucla.edu>

        * timespec.h (timespec_cmp): Avoid integer overflow
	by direct comparison instead of subtraction.

--- ../timespec_orig.h	2016-10-30 16:12:38.182266000 -0700
+++ lib/timespec.h	2016-10-30 16:14:38.956202000 -0700
@@ -45,8 +45,16 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec < b.tv_sec)
+    return -1;
+  else if (a.tv_sec > b.tv_sec)
+    return 1;
+  else if (a.tv_nsec < b.tv_nsec)
+    return -1;
+  else if (a.tv_nsec > b.tv_nsec)
+    return 1;
+  else
+    return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
